### PR TITLE
[BEAM-3873] Current version of commons-compress is DOS vulnerable CVE-2018-1324

### PR DESCRIPTION
**Pull request by iemejia**
DESCRIPTION HERE

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand:
   - [ ] What the pull request does
   - [ ] Why it does it
   - [ ] How it does it
   - [ ] Why this approach
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by iemejia** [comment]
retest this please

**Comment by iemejia** [comment]
retest this please

**Comment by iemejia** [comment]
Run Java Gradle PreCommit

File: pom.xml [hunk 1]
```diff
@@ -163,8 +163,8 @@
     <os-maven-plugin.version>1.5.0.Final</os-maven-plugin.version>
     <groovy-maven-plugin.version>2.0</groovy-maven-plugin.version>
     <license-maven-plugin.version>1.14</license-maven-plugin.version>
-    <maven-surefire-plugin.version>2.20.1</maven-surefire-plugin.version>
-    <maven-failsafe-plugin.version>2.20.1</maven-failsafe-plugin.version>
+    <maven-surefire-plugin.version>2.21.0</maven-surefire-plugin.version>
```
Comment (lukecwik) [inline pom.xml]: Did you mean to do this?

If so, can you update build.gradle to match this version:
https://github.com/apache/beam/blob/d8cb0e6098370dc5059834ea82c5f6fa22f6a91b/build.gradle#L159

Comment (iemejia) [reply pom.xml]: Yes, it was intended. I was not aware gradle was using that plugin too. Thanks for the info. I will fix that one too.

**Comment by iemejia** [comment]
Run Java Gradle PreCommit

File: pom.xml [hunk 2]
```diff
@@ -1460,6 +1461,7 @@
         <groupId>com.google.guava</groupId>
         <artifactId>guava-testlib</artifactId>
         <version>${guava.version}</version>
+        <type>test-jar</type>
         <scope>test</scope>
```
Comment (lukecwik) [inline pom.xml]: Why are we setting the `<type>` here (it doesn't seem to be consumed in any pom.xml)?

Also, we are duplicating the definition of guava-testlib around line 1400 in this pom.xml so we should be able to remove this entire definition.

File: pom.xml [hunk 3]
```diff
@@ -1446,6 +1446,7 @@
         <groupId>junit</groupId>
         <artifactId>junit</artifactId>
         <version>${junit.version}</version>
+        <type>test-jar</type>
```
Comment (lukecwik) [inline pom.xml]: We duplicate the junit definition around line 1400 as well so we should be able to remove this entire block here as well.

File: pom.xml (hunk 2 above)
Comment (iemejia) [reply pom.xml]: You are right, I thought the warning was related to the type and not the duplicated dependency. Removing it should make it.

File: pom.xml (hunk 3 above)
Comment (iemejia) [reply pom.xml]: Same as above.

**Comment by lukecwik** [comment]
Run Java PreCommit

**Pull request merged**